test/unit/local/steps/linux_producer.js

```diff
@@ -0,0 +1,184 @@
+          }))
+        ).eql([
+          {
+            action: 'scan',
+            kind: 'unknown',
+            path: `${mypath[0]}/${mypath[1]}`
+          }
+        ])
+        should(
+          (await producer.buffer.pop()).map(({ action, kind, path }) => ({
+            action,
+            kind,
+            path
+          }))
+        ).eql([
+          {
+            action: 'scan',
+            kind: 'unknown',
+            path: mypath.join('/')
+          }
+        ])
+      })
+    })
+
+    describe('register an event on FS events', () => {
+      let syncPath
+      let producer
+
+      before(async () => {
+        syncPath = fs.mkdtempSync(path.join(os.tmpdir(), 'foo-'))
+        producer = new LinuxProducer({
+          syncPath
+        })
+        await producer.start()
+        await producer.buffer.pop()
+      })
+
+      it('detect events on folder in temp dir', async () => {
+        const dirname = 'foobaz'
+        const newname = 'foobarbaz'
+        fs.mkdirSync(path.join(syncPath, dirname))
+        should(await producer.buffer.pop()).eql([
+          {
+            action: 'created',
+            kind: 'directory',
+            path: dirname
+          }
+        ])
+        fs.renameSync(
+          path.join(syncPath, dirname),
+          path.join(syncPath, newname)
+        )
+        should(await producer.buffer.pop()).eql([
+          {
+            action: 'renamed',
+            kind: 'directory',
+            oldPath: dirname,
+            path: newname
+          }
+        ])
+        fs.rmdirSync(path.join(syncPath, newname))
+        should(await producer.buffer.pop()).eql([
+          {
+            action: 'deleted',
+            kind: 'directory',
+            path: newname
+          }
+        ])
+      })
+
+      it('detect events on file in temp dir', async () => {
+        const filename = 'barbaz'
+        const newname = 'barfoobaz'
+        const content = 'Hello, Cozy Drive for Desktop'
+        fs.writeFileSync(path.join(syncPath, filename), content)
+        should(await producer.buffer.pop()).eql([
+          {
+            action: 'created',
+            kind: 'file',
+            path: filename
+          },
+          {
+            action: 'modified',
+            kind: 'file',
+            path: filename
+          }
+        ])
+        fs.renameSync(
+          path.join(syncPath, filename),
+          path.join(syncPath, newname)
+        )
+        should(await producer.buffer.pop()).eql([
+          {
+            action: 'renamed',
+            kind: 'file',
+            oldPath: 'barbaz',
+            path: 'barfoobaz'
+          }
+        ])
+        fs.unlinkSync(path.join(syncPath, newname))
+        should(await producer.buffer.pop()).eql([
+          {
+            action: 'deleted',
+            kind: 'file',
+            path: newname
+          }
+        ])
+      })
+    })
+  })
+}
+
+function mkdirSyncRecursive (syncPath = '/', mypath = '/') {
```

Comment:
`fs-extra` has the `ensureDir` method that does exactly that: https://github.com/jprichardson/node-fs-extra/blob/master/docs/ensureDir.md

```javascript
fse.ensureDir(path.join(syncPath, myPath))
```

Reply:
Starting from nodejs 10.2.0, the `mkdir` and `mkdirSync` functions from `fs` is also recursive. 10.x is a LTS. Do we plan to upgrade from 8.2.1? /poke @nono @sebn

Reply:
Our nodejs version is constrained by electron. We should upgrade electron, but it's often a burden, so I don't think we will do that in the next 15 days. After that, it depends of the priorities.